components/layout_2020/quotes.rs

```diff
@@ -0,0 +1,406 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+// Quotes data is obtained from ICU CLDR data file /tmp/cldr-common-46.0.zip.
+// TODO(xiaochengh): We should have some script to update the data automatically.
+// Or extend the icu create to provide the data directly.
+
+use std::collections::HashMap;
+
+use icu_locid::Locale;
+
+#[derive(Clone, Copy, Debug)]
+struct QuotesData(u16, u16, u16, u16);
+
+pub struct QuotePair {
+    pub opening: char,
+    pub closing: char,
+}
+
+static DEFAULT_QUOTES: QuotesData = QuotesData(0x201c, 0x201d, 0x2018, 0x2019);
+
+// TODO(xiaochengh): Make this map static or constructed only once
+fn get_quotes_map() -> HashMap<&'static str, QuotesData> {
+    let input = [
+        ("aa", QuotesData(0x201c, 0x201d, 0x2018, 0x2019)),
+        ("ab", QuotesData(0x201c, 0x201d, 0x2018, 0x2019)),
+        ("af", QuotesData(0x201c, 0x201d, 0x2018, 0x2019)),
```

Comment:
Please use a static `OnceLock` here to avoid building this map more than once. In addition, I think we should be using real Unicode characters here instead of constantly building the quotes from components parts. This can be done eventually with an import script or I'm happy to help out here converting the data.

Reply:
Done. Thanks for fixing my Rust noobie issues.